File: leetcode67.cpp
class Solution {
public:
    string addBinary(string a, string b) {
        int c = 0;
        string res = "";
        int len1 = a.size() -1;
        int len2 = b.size() -1;
        while(len1 >=0 || len2 >=0 || c == 1)
        {
        	c +=  len1 >=0 ? a[len1--] - '0': 0;
        	c += len2 >=0 ? b[len2--] - '0': 0;
        	res = char(c % 2 + '0') + res;
        	c /= 2;
        }
        return res;
    }
};

File: leetcode67.java
public class Solution {
    public String addBinary(String a, String b) {
        int len1 = a.length()-1;
        int len2 = b.length()-1;
        int c = 0;
        StringBuffer sb = new StringBuffer();
        while(len1 >= 0 || len2 >= 0){
        	int tmp = c;
        	if(len1 >= 0){
        		tmp += a.charAt(len1--) - '0';
        	}
        	if(len2 >= 0){
        		tmp += b.charAt(len2--) - '0';
        	}
        	sb.append(tmp%2);
        	c = tmp / 2;
        }
        if(c != 0)
        	sb.append(c);
        return sb.reverse().toString();

    }
}
